src/main/webapp/app/exercises/shared/feedback/group/programming-feedback-groups.ts

```diff
@@ -0,0 +1,87 @@
+import { FeedbackGroup } from 'app/exercises/shared/feedback/group/feedback-group';
+import { FeedbackItem } from 'app/exercises/shared/feedback/item/feedback-item';
+import { Exercise } from 'app/entities/exercise.model';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+
+/**
+ * Returns all FeedbackItemGroups for Programming exercises in the order, in which they will be displayed
+ */
+export const getAllFeedbackGroups = (exercise: Exercise): FeedbackGroup[] => {
+    return [new FeedbackGroupWrong(), new FeedbackGroupWarning(exercise), new FeedbackGroupInfo(), new FeedbackGroupCorrect()];
+};
+
+/**
+ * Negative feedbacks that are not SCA
+ */
+class FeedbackGroupWrong extends FeedbackGroup {
+    constructor() {
+        super();
+        this.name = 'wrong';
+        this.color = 'danger';
+        this.open = true;
+    }
+
+    shouldContain(feedbackItem: FeedbackItem): boolean {
+        const isReviewerFeedback = feedbackItem.type === 'Reviewer' && feedbackItem.credits && feedbackItem.credits < 0;
```

Comment:
Why not use the proper Enums here?

Reply:
Typescript enums have strange behaviours. i.e. inspecting in dev tools leads to `feedbackItem.type: 0` because enums after compilation are just numbers.
https://www.youtube.com/watch?v=0fTdCSH_QEU